cranelift/fuzzgen/src/lib.rs

```diff
@@ -224,14 +289,14 @@ where
         Ok(ctx.func)
     }
 
-    fn generate_func(&mut self) -> Result<Function> {
+    pub fn generate_func(&mut self) -> Result<Function> {
         let func = FunctionGenerator::new(&mut self.u, &self.config).generate()?;
         self.run_func_passes(func)
```

Comment:
`run_func_passes` builds a `TargetIsa` which probably doesn't match `self.isa`. I think the passes it runs probably don't actually look at anything except the target-independent flags, so that's probably fine, but I'm not sure. I think I've been confused about this before and forgotten the answer. Maybe @afonso360 knows?

Reply:
Yeah it does. We only need to build it because `canonicalize_nans` requires it, although it doesn't use it besides the `enable_verifier` flag.

It would remove quite a bit of code to not have to build it again, but it would be nice if we could avoid verifying it there, since I benchmarked it once and it ended up showing on the profiling. (We still run the verifier while compiling)

The other passes don't use the target ISA.